src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/Integrations/ConsumerConsumeSyncIntegration.cs

```diff
@@ -46,13 +46,16 @@ internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, int tim
             consumeResult = response == null ? null : response.DuckAs<IConsumeResult>();
         }
 
-        var activity = KafkaInstrumentation.StartConsumerActivity(consumeResult, (DateTimeOffset)state.StartTime!, instance!);
-        if (exception is not null)
+        if (consumeResult is not null)
         {
-            activity.SetException(exception);
-        }
+            var activity = KafkaInstrumentation.StartConsumerActivity(consumeResult, (DateTimeOffset)state.StartTime!, instance!);
+            if (exception is not null)
```

Comment:
may hide some transient network errors (since `consumeResult' has to exist) ... not sure if good or bad

Reply:
I'd leave it as is for the quick fix.

Reply:
Isn't that guaranteed by the check on line 40?